R/fcast.R

```diff
@@ -34,7 +34,37 @@ check_formula = function(formula, varnames, valnames) {
   allvars = c(vars, valnames)
   if (any(allvars %chin% varnames[duplicated(varnames)]))
     stopf('data.table to cast must have unique column names')
-  deparse_formula(as.list(formula)[-1L], varnames, allvars)
+  if (value.var.in.LHSdots == value.var.in.RHSdots && isFALSE(value.var.in.LHSdots)) {
+    deparse_formula(as.list(formula)[-1L], varnames, allvars)
+  }
+  else if (value.var.in.LHSdots == value.var.in.RHSdots && isTRUE(value.var.in.LHSdots)) {
+    vars = setdiff(vars, valnames)
+    split_deparsing(as.list(formula)[-1L], varnames, vars, vars)
+  }
+  else if (isTRUE(value.var.in.LHSdots) && isFALSE(value.var.in.RHSdots)) {
+    vars = setdiff(vars, valnames)
+    split_deparsing(as.list(formula)[-1L], varnames, vars, allvars)
+  }
+  else {
+    vars = setdiff(vars, valnames)
+    split_deparsing(as.list(formula)[-1L], varnames, allvars, vars)
```

Comment:
The only difference between these three `split_deparsing` cases is vars,vars vs vars,allvars vs allvars,vars right?
In that case can you please refactor the code to remove the repetition?
Also can the case above with `deparse_formula` be considered a special case?

Reply:
I answer first your second comment. My first attempt was a unique function instead of "duplicating" `deparse_formula` with its variant `split_deparsing`, But I didn't get it to work.

Regarding the difference between the three `split_deparsing` cases, for the moment the solutions I can think of move such code complexity to another place. Well, I can write a list for each of both of the last arguments of `split_deparsing` and call this function depending on the list name. I may try and commit.

Reply:
Done!